src/main/scala/org/neo4j/spark/service/Neo4jQueryService.scala

```diff
@@ -1,39 +1,77 @@
 package org.neo4j.spark.service
 
 import org.apache.spark.sql.SaveMode
-import org.apache.spark.sql.sources.{And, EqualTo, Filter, IsNull, Not, Or}
-import org.neo4j.cypherdsl.core.{Condition, Conditions, Cypher, Functions, PropertyContainer, StatementBuilder}
+import org.apache.spark.sql.sources.{And, Filter, Or}
 import org.neo4j.cypherdsl.core.renderer.Renderer
-import org.neo4j.spark.{Neo4jOptions, QueryType}
+import org.neo4j.cypherdsl.core._
 import org.neo4j.spark.util.Neo4jImplicits._
 import org.neo4j.spark.util.Neo4jUtil
+import org.neo4j.spark.{Neo4jOptions, NodeSaveMode, QueryType}
 
-import collection.JavaConverters._
+import scala.collection.JavaConverters._
 
 class Neo4jQueryWriteStrategy(private val saveMode: SaveMode) extends Neo4jQueryStrategy {
   override def createStatementForQuery(options: Neo4jOptions): String =
-    s"""UNWIND ${"$"}events AS event
+    s"""UNWIND ${"$"}events AS $BATCH_VARIABLE
        |${options.query.value}
        |""".stripMargin
 
-  override def createStatementForRelationships(options: Neo4jOptions): String = throw new UnsupportedOperationException("TODO implement method")
+  private def keywordFromSaveMode(saveMode: Any): String = {
+    saveMode match {
+      case NodeSaveMode.Overwrite | SaveMode.Overwrite => "MERGE"
```

Comment:
See earlier comment.   "MERGE" would make more sense than overwrite

Reply:
As I said here https://github.com/neo4j-contrib/neo4j-spark-connector/pull/169#discussion_r481907157 we tried to stick to Spark syntax, because we are constrained to use them via the `mode` API, so we tried to reuse the seme semantics also here

Reply:
ok; understood.  I can accept this, I'm more of a spark noob